cmd/bbolt/command_surgery_cobra.go

```diff
@@ -128,3 +135,74 @@ func surgeryFreelistAbandonFunc(cmd *cobra.Command, args []string) error {
 	fmt.Fprintf(os.Stdout, "The freelist was abandoned in both meta pages.\nIt may cause some delay on next startup because bbolt needs to scan the whole db to reconstruct the free list.\n")
 	return nil
 }
+
+func newSurgeryFreelistRebuildCommand() *cobra.Command {
+	rebuildFreelistCmd := &cobra.Command{
+		Use:   "rebuild <bbolt-file> [options]",
+		Short: "Rebuild the freelist",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("db file path not provided")
+			}
+			if len(args) > 1 {
+				return errors.New("too many arguments")
+			}
+			return nil
+		},
+		RunE: surgeryFreelistRebuildFunc,
+	}
+
+	rebuildFreelistCmd.Flags().StringVar(&surgeryTargetDBFilePath, "output", "", "path to the target db file")
+
+	return rebuildFreelistCmd
+}
+
+func surgeryFreelistRebuildFunc(cmd *cobra.Command, args []string) error {
+	srcDBPath := args[0]
+
+	// Ensure source file exists.
+	fi, err := os.Stat(srcDBPath)
+	if os.IsNotExist(err) {
```

Comment:
I ran the command on my local env and felt that error message was a little bit hard to interpret, how about something like this?
```
	if os.IsNotExist(err) {
		return fmt.Errorf("Source database file %s doesn't exist.", srcDBPath)
	} else if err != nil {
		return fmt.Errorf("Failed to open source database file %s: %s", srcDBPath, err)
	}

	if surgeryTargetDBFilePath == "" {
		return fmt.Errorf("Output path wasn't given, specify output database file path with --output option.")
	}
```

Reply:
thx & updated.